contracts/reserve-auction/src/helpers.rs

```diff
@@ -85,92 +21,100 @@ pub fn only_no_auction(deps: Deps, collection: &Addr, token_id: &str) -> Result<
     Ok(())
 }
 
-pub fn load_marketplace_params(
-    deps: Deps,
-    marketplace_addr: &Addr,
-) -> Result<ParamsResponse, ContractError> {
-    let marketplace_params: ParamsResponse = deps
-        .querier
-        .query_wasm_smart(marketplace_addr, &MarketplaceQueryMsg::Params {})?;
-    Ok(marketplace_params)
+pub fn validate_reserve_price(
+    storage: &dyn Storage,
+    check_reserve_price: &Coin,
+) -> Result<(), ContractError> {
+    let minimum_amount = MIN_RESERVE_PRICES.may_load(storage, check_reserve_price.denom.clone())?;
+
+    ensure!(
+        minimum_amount.is_some(),
+        ContractError::InvalidInput("invalid reserve price denom".to_string(),)
+    );
+
+    ensure!(
+        check_reserve_price.amount >= minimum_amount.unwrap(),
+        ContractError::InvalidReservePrice {
+            min: coin(
+                minimum_amount.unwrap().u128(),
+                check_reserve_price.denom.clone(),
+            ),
+        }
+    );
+
+    Ok(())
 }
 
 pub fn settle_auction(
```

Comment:
I wouldn't consider this function a "helper". If you want it separate from `execute.rs` maybe it can be in a file called `abci.rs` or `settle.rs`.

Reply:
I agree, but I just need to call it from both `sudo.rs` and `execute.rs`